src/core/listen.hpp

```diff
@@ -48,56 +49,11 @@
 
 enum class EnableTransparentMode : bool { No, Yes };
 
-inline unique_fd create_local_server(char const* sck_name)
-{
-    unique_fd unique_sck {socket(PF_INET, SOCK_STREAM, 0)};
-    int const sck = unique_sck.fd();
-
-    /* reuse same port if a previous daemon was stopped */
-    int allow_reuse = 1;
-    setsockopt(sck, SOL_SOCKET, SO_REUSEADDR, &allow_reuse, sizeof(allow_reuse));
-
-    /* set snd buffer to at least 32 Kbytes */
-    int snd_buffer_size = 32768;
-    unsigned int option_len = sizeof(snd_buffer_size);
-    if (0 == getsockopt(sck, SOL_SOCKET, SO_SNDBUF, &snd_buffer_size, &option_len)) {
-        if (snd_buffer_size < 32768) {
-            snd_buffer_size = 32768;
-            setsockopt(sck, SOL_SOCKET, SO_SNDBUF, &snd_buffer_size, sizeof(snd_buffer_size));
-        }
-    }
-
-    /* set non blocking */
-    fcntl(sck, F_SETFL, fcntl(sck, F_GETFL) | O_NONBLOCK);
-
-    union
-    {
-        sockaddr_un s;
-        sockaddr addr;
-    } u;
-
-    auto len = strnlen(sck_name, sizeof(u.s.sun_path)-1u);
-    memcpy(u.s.sun_path, sck_name, len);
-    u.s.sun_path[len] = 0;
-    u.s.sun_family = AF_UNIX;
-
-    LOG(LOG_INFO, "Listen: binding socket %d on %s", sck, sck_name);
-    if (0 != ::bind(sck, &u.addr, sizeof(u))) {
-        LOG(LOG_ERR, "Listen: error binding socket [errno=%d] %s", errno, strerror(errno));
-        return invalid_fd();
-    }
-
-    LOG(LOG_INFO, "Listen: listening on socket %d", sck);
-    if (0 != listen(sck, 2)) {
-        LOG(LOG_ERR, "Listen: error listening on socket");
-    }
-
-    // OK, keep the temporary socket everything was fine
-    return unique_sck;
-}
-
-inline unique_fd create_server_bind_sck(
-    unique_fd fd_sck, sockaddr& addr, EnableTransparentMode enable_transparent_mode)
+inline unique_fd create_server_bind_sck
+(unique_fd fd_sck,
+ sockaddr_storage& addr,
+ socklen_t addrlen,
+ EnableTransparentMode enable_transparent_mode) noexcept
```

Comment:
style: :vomiting_face:

Reply:
It's for avoid to exceed 80 columns for readability but I understand well your disagreement :/

Reply:
Looking forward for the code style guide ;)